tuxemon/monster.py

```diff
@@ -528,7 +514,7 @@ def set_level(self, level: int = 5) -> None:
 
         # Update moves
         for move in self.moveset:
-            if move not in self.moves and move.level_learned <= level:
```

Comment:
My point was that `move not in self.moves` is meant to stop you from learning moves you already know. It's possible that python isn't realising that they are the same move, because we don't have object equality defined properly. So, you might need to do:

`if move.slug not in (m.slug for m in self.moves) and move.level_learned <= level`

Reply:
It was easier than I thought. I was thinking about removing duplicates or getting common elements inside the two lists. Thanks